src/components/network/network-table.js

```diff
@@ -50,10 +71,136 @@ genotet.NetworkTable.prototype.create = function(table, edges) {
       {title: 'Weight', data: 'weight'},
       {title: '', data: 'added'}
     ],
+    select: {
+      style: 'os',
+      info: false
+    },
+    dom: '<"row"<"#table-btn.col-sm-12"B>>' +
+      '<"row">' +
+      '<"row"<"col-sm-5"l><"col-sm-7"f>>' +
+      '<"row"<"col-sm-12"tr>>' +
+      '<"row"<"col-sm-5"i><"col-sm-7"p>>',
+    buttons: [
+      {
+        text: 'Add',
+        className: 'btn btn-default',
+        action: function(e, dt, node, config) {
+          var selectedEdges = dt.rows({selected: true}).data();
+          var additionEdges = [];
+          for (var i = 0; i < selectedEdges.length; i++) {
+            additionEdges.push({
+              id: selectedEdges[i].id,
+              source: selectedEdges[i].source,
+              target: selectedEdges[i].target,
+              weight: selectedEdges[i].originalWeight
+            });
+            edgesForTable.forEach(function(edge) {
+              if (selectedEdges[i].id == edge.id) {
+                edge.added = true;
+              }
+            });
+          }
+          this.signal('addEdges', {
+            edges: additionEdges
+          });
+
+          // refresh the row
+          dt.rows({selected: true}).invalidate();
+          // change the button status
+          dt.buttons(0).enable(false);  // the addition button
+          dt.buttons(1).enable(true);   // the removal button
+        }.bind(this),
+        enabled: false
+      },
+      {
+        text: 'Remove',
+        className: 'btn btn-default',
+        action: function(e, dt, node, config) {
+          var selectedEdges = dt.rows({selected: true}).data();
+          var removalEdges = [];
+          for (var i = 0; i < selectedEdges.length; i++) {
+            removalEdges.push({
+              id: selectedEdges[i].id,
+              source: selectedEdges[i].source,
+              target: selectedEdges[i].target,
+              weight: selectedEdges[i].originalWeight
+            });
+            edgesForTable.forEach(function(edge) {
+              if (selectedEdges[i].id == edge.id) {
+                edge.added = false;
+              }
+            });
+          }
+          this.signal('highlightEdges', {
+            edgesId: []
+          });
+          this.signal('removeEdges', {
+            edges: removalEdges
+          });
+          this.signal('hideEdge', {
+            edges: removalEdges
+          });
+          // refresh the row
+          dt.rows({selected: true}).invalidate();
+          // change the button status
+          dt.buttons(0).enable(true);   // the addition button
+          dt.buttons(1).enable(false);  // the removal button
+        }.bind(this),
+        enabled: false
+      }
+    ],
     lengthMenu: [5, 10, 20, 50],
     pageLength: 5,
     pagingType: 'full'
   });
+
+  table.on('click', function() {
+    var data = dataTable.rows({selected: true}).data();
```

Comment:
What are "a lot of attributes"? It seems that you are accessing attributes such as "added". It would be clearer to add type annotations for the data object here, at least for what you would be accessing.